### PR TITLE
Data Builder Record Abstraction

### PR DESCRIPTION
**Pull request by AndrewCiambrone**

### CODE REVIEW

**Comment by feng-tao** [comment]
will take a look, thanks for the RFC

File: templates/000-databuilder-record-abstraction.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+- Feature Name: Add a abstraction layer to the databuilder's record model
+- Start Date: 2020-10-02
+- RFC PR: [amundsen-io/rfcs#0000](https://github.com/amundsen-io/rfcs/pull/0000) (after opening the RFC PR, update this with a link to it and update the file name)
```
Comment (Golodhros) [inline templates/000-databuilder-record-abstraction.md:3]:
```suggestion
- RFC PR: [amundsen-io/rfcs#5](https://github.com/amundsen-io/rfcs/pull/5) (after opening the RFC PR, update this with a link to it and update the file name)
```

**Review by feng-tao — COMMENTED** [review]
do you see any backward compatibility issue?

**Comment by feng-tao** [comment]
also the DCO fails as well

File: templates/000-databuilder-record-abstraction.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- reverse_label: str
+- attributes: Dict
+
+The loader would be then be responsible for taking these graph nodes and graph relationships and converting them to the format expected by the publisher. 
+
+The unpolished changes mentioned above can be found: https://github.com/AndrewCiambrone/amundsendatabuilder/pull/6
+
+## Drawbacks
+
+This adds a level of complexity because it adds an additional layer to the databuilder. Instead of having the neo4j serialization built into the models it becomes a separate process. This also would add a small performance degradation because the models are producing named tuples and then they are being convered into another format.
```
Comment (feng-tao) [inline templates/000-databuilder-record-abstraction.md:52]: could you elaborate a bit or maybe you could create the pr and point me to the place of this?

**Comment by feng-tao** [comment]
also could you create the pr with your branch? thanks a lot

**Comment by AndrewCiambrone** [comment]
Pr: https://github.com/amundsen-io/amundsendatabuilder/pull/380

**Comment by AndrewCiambrone** [comment]
> do you see any backward compatibility issue?

I am not sure if its a bug or a typo but some of the timestamps were not unquoted. I currently have it so it is unquoted let me know if that is unwanted.

File: templates/000-databuilder-record-abstraction.md (hunk 2 above)
Comment (feng-tao) [reply templates/000-databuilder-record-abstraction.md:52]: ah I see, it should be fine.

**Comment by feng-tao** [comment]
@AndrewCiambrone I just merged the change. Could you update this rfc and move it under rfcs folder and name it 002-xxx.md? thanks.

**Review by feng-tao — APPROVED** [review]

**Pull request merged**